contrib/graphite/blueflood.py

```diff
@@ -43,37 +81,105 @@ def __init__(self, config=None):
 
     self.tenant = tenant
     self.bf_query_endpoint = urls[0]
+    self.enable_submetrics = enable_submetrics
+    self.submetric_aliases = submetric_aliases
+    self.client = BluefloodClient(self.bf_query_endpoint, self.tenant, 
+                                  self.enable_submetrics, self.submetric_aliases)
+    print("BF finder submetrics enabled: ", enable_submetrics)
+
+  def complete(self, metric, query_depth):
+    #is submetric name complete?
+    return len(metric.split('.')) == (query_depth - 1)
+
+  def make_request(self, url, payload, headers):
+    if auth.is_active():
+      headers['X-Auth-Token'] = auth.get_token(False)
+    r = requests.get(url, params=payload, headers=headers)
+    if r.status_code == 401 and auth.is_active():
+      headers['X-Auth-Token'] = auth.get_token(True)
+      r = requests.get(url, params=payload, headers=headers)
+    return r
 
+  def find_metrics_endpoint(self, endpoint, tenant):
+    return "%s/v2.0/%s/metrics/search" % (endpoint, tenant)
+
+  def find_metrics(self, query):
+    print "BluefloodClient.find_metrics: " + str(query)
+    payload = {'query': query}
+    headers = auth.headers()
+    endpoint = self.find_metrics_endpoint(self.bf_query_endpoint, self.tenant)
+    r = self.make_request(endpoint, payload, headers)
+
+    if r.status_code == 200:
+        return [m['metric'] for m in r.json()]
+    else:
```

Comment:
People have been using it without a problem I think, and I've had to work on so many other issues that I'm not going to do this one yet.